c/main.c

```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/sha.h>
+
+unsigned int hexstr2bytes(unsigned char *hex_str, unsigned char *out_buff){
+	int i, n;
+	int length = strlen(hex_str) / 2;
+
+	for(i = 0; i < length; i++) {
+		sscanf(hex_str, "%2X", &n);
+		hex_str += 2;
+		out_buff[i] = (char)n;
+	}
+	return length;
+}
+
+
+int main(int argc, char **argv) {
+	unsigned char comp_str[64];
+	unsigned char email[128];
+	unsigned char hex_byte[16];
+	unsigned char hash[SHA_DIGEST_LENGTH];
+	unsigned int i = 0;
+	unsigned int comp_length = 0;
+
+	if ( argc != 4 ) {
+
+		printf( "Usage: %s yourname pattern domain\n", argv[0] );
+		printf( "Example: %s joe c0ffee gmail.com\n", argv[0] );
+		return -1;
+
+	} else {
+		strcpy(comp_str, argv[1]);
+		strcat(comp_str, ".%d@");
```

Comment:
Hey thanks @nenadilic84 for the pull request. Is `".%d@"` making `user.123@example.com`? I ask because generally it will generate a different email than the one they provided. Try looking into using a `+` instead of `.` because most hosts will route it to a label or folder.

https://support.google.com/mail/answer/12096?hl=en

:D